packages/editor-ui/src/modules/nodeTypes.ts

```diff
@@ -0,0 +1,87 @@
+import Vue from 'vue';
+import { ActionContext, Module } from 'vuex';
+import type { INodeTypeDescription } from 'n8n-workflow';
+
+import type { IRootState, INodeTypesState } from '../Interface';
+import { DEFAULT_NODETYPE_VERSION } from '@/constants';
+import { getNodeTypes } from '@/api/nodeTypes';
+
+const module: Module<INodeTypesState, IRootState> = {
+	namespaced: true,
+	state: {
+		nodeTypes: {},
+	},
+	getters: {
+		allNodeTypes: (state): INodeTypeDescription[] => {
+			return Object.values(state.nodeTypes);
+		},
+		nodeType: (state) => (nodeType: string, version?: number): INodeTypeDescription | null => {
```

Comment:
since it returns a function, we should name this `getNodeType` or `getNodeTypeByName`..  also maybe param can be `nodeTypeName`